python/cuml/test/test_module_config.py

```diff
@@ -0,0 +1,114 @@
+#
+# Copyright (c) 2019, NVIDIA CORPORATION.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import pytest
+
+import cudf
+import cuml
+import cupy as cp
+import numpy as np
+
+from numba.cuda import is_cuda_array, as_cuda_array
+
+global_input_types = [
+    'numpy', 'numba', 'cupy', 'cudf'
+]
+
+test_output_types = {
+    'numpy': np.ndarray,
+    'cupy': cp.ndarray,
+    'cudf': cudf.Series
+}
+
+
+@pytest.mark.parametrize('input_type', global_input_types)
+@pytest.mark.filterwarnings('ignore::UserWarning')
+def test_default_global_output_type(input_type):
+    dataset = get_small_dataset(input_type)
+
+    dbscan_float = cuml.DBSCAN(eps=1.0, min_samples=1)
+    dbscan_float.fit(dataset)
+
+    res = dbscan_float.labels_
+
+    if input_type == 'numba':
+        assert is_cuda_array(res)
+    else:
+        assert isinstance(res, test_output_types[input_type])
+
+
+@pytest.mark.parametrize('global_type', global_input_types)
+@pytest.mark.parametrize('input_type', global_input_types)
+def test_global_output_type(global_type, input_type):
+    dataset = get_small_dataset(input_type)
+
+    cuml.set_global_output_type(global_type)
```

Comment:
I'm always concerned when I see globals. We should consider, instead, using a cascading configuration approach (`Commons-configuration` comes to mind from my days in the Java world, would that be `configparser` in the Python world?) that will prioritize options based on their distance from the user (and provide a centralized API to retrieve an option by just saying "config.get(optionName, default_value). This will alleviate us from having to set globals (the default becomes the `global`.